diff -urN CMakeLists.txt.old CMakeLists.txt
--- CMakeLists.txt.old	2016-04-06 10:41:53.000000000 +0200
+++ CMakeLists.txt	2016-04-06 10:45:37.620848467 +0200
@@ -50,19 +50,10 @@
 	message(FATAL_ERROR "Can't find any of the files /usr/include/readline/readline.h")
 endif()
 
-if ( (NOT EXISTS "/usr/bin/pandoc"))
-	message(FATAL_ERROR "Can't find any of the files /usr/bin/pandoc")
-endif()
-
 if ( (NOT EXISTS "/usr/bin/curl"))
 	message(FATAL_ERROR "Can't find any of the files /usr/bin/curl")
 endif()
 
-find_program ( WHERE_PANDOC pandoc )
-if ( NOT WHERE_PANDOC )
-	MESSAGE(FATAL_ERROR "Error! PANDOC is not installed.")
-endif()
-
 add_subdirectory(doc)
 add_subdirectory(data/debian)
 add_subdirectory(po)
