diff -urN CMakeLists.txt.old CMakeLists.txt
--- CMakeLists.txt.old	2015-07-20 20:58:56.000000000 +0200
+++ CMakeLists.txt	2015-07-20 21:00:02.712400671 +0200
@@ -49,19 +49,10 @@
 	message(FATAL_ERROR "Can't find the file /usr/include/readline/readline.h")
 endif()
 
-if ( NOT EXISTS "/usr/bin/pandoc" )
-	message(FATAL_ERROR "Can't find the file /usr/bin/pandoc")
-endif()
-
 if ( NOT EXISTS "/usr/bin/curl" )
 	message(FATAL_ERROR "Can't find the file /usr/bin/curl")
 endif()
 
-find_program ( WHERE_PANDOC pandoc )
-if ( NOT WHERE_PANDOC )
-	MESSAGE(FATAL_ERROR "Error! PANDOC is not installed.")
-endif()
-
 add_subdirectory(doc)
 add_subdirectory(data/debian)
 add_subdirectory(po)
