diff -urN CMakeLists.txt.old CMakeLists.txt
--- CMakeLists.txt.old	2016-11-11 11:25:09.457957712 +0000
+++ CMakeLists.txt	2016-11-11 11:26:11.455954148 +0000
@@ -52,18 +52,10 @@
 if ( (NOT EXISTS "/usr/include/readline/readline.h"))
 	message(FATAL_ERROR "Can't find any of the files /usr/include/readline/readline.h")
 endif()
-if ( (NOT EXISTS "/usr/bin/pandoc"))
-	message(FATAL_ERROR "Can't find any of the files /usr/bin/pandoc")
-endif()
 if ( (NOT EXISTS "/usr/bin/curl"))
 	message(FATAL_ERROR "Can't find any of the files /usr/bin/curl")
 endif()
 
-find_program ( WHERE_PANDOC pandoc )
-if ( NOT WHERE_PANDOC )
-	MESSAGE(FATAL_ERROR "Error! PANDOC is not installed.")
-endif()
-
 find_program ( WHERE_GRESOURCE glib-compile-resources )
 if ( NOT WHERE_GRESOURCE )
 	MESSAGE(FATAL_ERROR "Error! GLIB-COMPILE-RESOURCES is not installed.")
