diff -urN usr/share/X11/xkb/rules/base.old usr/share/X11/xkb/rules/base
--- usr/share/X11/xkb/rules/base.old	2008-03-24 21:29:00.000000000 +0100
+++ usr/share/X11/xkb/rules/base	2008-03-24 21:32:35.000000000 +0100
@@ -101,6 +101,7 @@
               genius geniuscomfy2 \
               gyration honeywell_euroboard \
               hp250x hp5xx hp500fa hpdv5 \
+              aspire5033 \
               hpi6 hpxe3gc hpxe3gf hpxe4xxx hpxt1000 hpzt11xx htcdream \
               ibm_spacesaver ipaq inspiron intel \
               logiaccess logicda logicink \
